code/modules/clothing/neck/_neck.dm

```diff
@@ -452,6 +452,10 @@
 
 /obj/item/clothing/neck/petcollar/attack_self(mob/user)
 	tagname = sanitize_name(tgui_input_text(user, "Would you like to change the name on the tag?", "Pet Naming", "Spot", MAX_NAME_LEN))
+	if (!tagname || !length(tagname))
```

Comment:
While you're here you should filter out pure whitespace names as well

Reply:
reject_bad_name in sanitize_name already checks for it, requiring at least 2 alphanumerics to return a string